go-controller/pkg/ovn/controller/apbroute/network_client.go

```diff
@@ -204,8 +205,9 @@ func (nb *northBoundClient) addGatewayIPs(pod *v1.Pod, egress *gateway_info.Gate
 		podIPs = append(podIPs, ipNet)
 	}
 	if len(podIPs) == 0 {
-		klog.Warningf("Will not add gateway routes pod %s/%s. IPs not found!", pod.Namespace, pod.Name)
-		return nil
+		// At this stage the pod is either in Pending or Running phase, but Pending should not have an IP, therefore it should not
```

Comment:
maybe as an extra check you should do:

```
if len(podIPs) == 0 && !util.PodCompleted(pod)
```

Reply:
in line [193](https://github.com/ovn-org/ovn-kubernetes/pull/3934/files#diff-36c7388a83f3572d73d68a771bcd20ea0adabe13b7aad9c8b500b398ba57863aL193) it already checks if the pod is completed.